apps/web/src/app/(private)/settings/integrations/[integrationId]/destroy/page.tsx

```diff
@@ -17,6 +67,8 @@ export default function DestroyIntegration({
   const { data, destroy, isDestroying } = useIntegrations()
   const integration = data.find((p) => p.id === Number(integrationId))
 
+  const { data: references, isLoading } = useIntegrationReferences(integration)
```

Comment:
Wait wait. Also this. You're forced to implement a full store just because this nextjs page is client side. Remove the `use client`

Now this will a server page. 

1. Fetch integrations
2. Fetch integrations references 

Both on server side. 

Then you show Warning modal if references exists

Otherwise show the destroy modal. No need to change the behaviour and implement `children` prop. Destroy modal is simple for a reason. 

To handle the client on click you just need to move the Destroy modal in another component where `use client` is set.

Setting `use client` in nextjs `page.tsx` is anti-pattern